processing/src/main/java/org/apache/druid/segment/nested/VariantColumn.java

```diff
@@ -310,6 +312,32 @@ public int lookupId(String val)
     return -1;
   }
 
+
+  public IntSet lookupIds(String val)
+  {
+    IntSet intList = new IntArraySet(3);
+    if (val == null) {
+      intList.add(0);
+      return intList;
+    }
+    int candidate = stringDictionary.indexOf(StringUtils.toUtf8ByteBuffer(val));
+    if (candidate >= 0) {
+      intList.add(candidate);
+    }
+    candidate = longDictionary.indexOf(GuavaUtils.tryParseLong(val));
+    if (candidate >= 0) {
+      candidate += adjustLongId;
+      intList.add(candidate);
+    }
+    candidate = doubleDictionary.indexOf(Doubles.tryParse(val));
```

Comment:
we have this pattern in a few other places where we are trying to find the index/indexes for some value, i just copied it, but yeah, can fix since only the first dictionary can have null

Reply:
fixed